src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -409,6 +409,25 @@ public override bool IsGridVisible
                 SetGridVisibility();
             }
         }
+        
+        /// <summary>
+        /// Identifies if the Graph yields any rendered geometry
+        /// Any graph would always render at least 3 elements:
+        /// Headlight, Grid, and Axis
+        /// Anything more than the default elements is considered rendered geometry
+        /// </summary>
+        public bool HasRenderedGeometry
+        {
+            get
+            {
+                return Element3DDictionary.Count(g => 
```

Comment:
1. this seems quite wasteful in terms of performance, but it's unclear to me from this PR what the anticipated use of this API is - can you give some context?

2. In addition, it appears one could call this API in and get incorrect data, ie, when is it safe to call this API? Is there an event that it should be called after, a specific task in the scheduler? From the UI thread? After a run? This should be included in the summary.

3. Do you consider geometry with preview off to be rendered? Maybe add a test for that case and describe the behavior in the summary.

Reply:
You are right, I did not think about that. My initial attempt was a simple `dictionary.Count > 3 `check, which was a bit vague. I thought this is more explicit, however, I realize it is quite computationally intensive for large graphs. 

It seems we always draw 3 default elements, which constitutes an 'empty' scene. 

I did think of hidden geometry, but in my use case, this still counts as an empty scene. 

Currently, there is no way to access this information from the context of an external extension hence the implementation of this property. I can revert back to the simpler check, which should address the performance concern I hope. In terms of scope, the intention is to query after a run has finished and the dispatcher has no remaining tasks. I can add this to the description.

Reply:
well the context is important, if this method is called infrequently than maybe it does not matter - though iterating every single geometry object could be very slow depending on what is being drawn.

Reply:
I pushed the changed version, hopefully, this is a bit clearer and less misleading to use.